<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    
    <title>gatk_snp流程.md | sqw workplace</title>
    
    
        <meta name="keywords" content="gatk_snp流程.md">
    
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="description" content="gatk官方博文，描述 gatk4和之前版本的差异 http:&#x2F;&#x2F;broad.io&#x2F;GATK4-1一些terra上的实例：https:&#x2F;&#x2F;terra.bio&#x2F;#library&#x2F;showcase 数据预处理：map    bam文件中会包含read group的信息，在之后的分析中要用到mark duplicatesrecalibrate base quality score（BQSR）：对mach">
<meta property="og:type" content="article">
<meta property="og:title" content="gatk_snp流程.md">
<meta property="og:url" content="https://sqwwww.github.io/wiki/pipeline/gatk_snp%E6%B5%81%E7%A8%8B/index.html">
<meta property="og:site_name" content="sqw workplace">
<meta property="og:description" content="gatk官方博文，描述 gatk4和之前版本的差异 http:&#x2F;&#x2F;broad.io&#x2F;GATK4-1一些terra上的实例：https:&#x2F;&#x2F;terra.bio&#x2F;#library&#x2F;showcase 数据预处理：map    bam文件中会包含read group的信息，在之后的分析中要用到mark duplicatesrecalibrate base quality score（BQSR）：对mach">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2022-07-16T09:01:36.000Z">
<meta property="article:modified_time" content="2022-09-29T03:40:29.930Z">
<meta property="article:author" content="sqw">
<meta name="twitter:card" content="summary">
    

    
        <link rel="alternate" href="/atom.xml" title="sqw workplace" type="application/atom+xml">
    

    
        <link rel="icon" href="/favicon.ico">
    

    
<link rel="stylesheet" href="/libs/font-awesome/css/font-awesome.min.css">

    
<link rel="stylesheet" href="/libs/open-sans/styles.css">

    
<link rel="stylesheet" href="/libs/source-code-pro/styles.css">


    
<link rel="stylesheet" href="/css/style.css">

    
<script src="/libs/jquery/2.1.3/jquery.min.js"></script>

    
<script src="/libs/jquery/plugins/cookie/1.4.1/jquery.cookie.js"></script>

    
    
        
<link rel="stylesheet" href="/libs/lightgallery/css/lightgallery.min.css">

    
    
        
<link rel="stylesheet" href="/libs/justified-gallery/justifiedGallery.min.css">

    
    
    
    


    
        <script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    
<meta name="generator" content="Hexo 6.2.0"></head>

<body>
    <div id="container">
        <header id="header">
    <div id="header-main" class="header-inner">
        <div class="outer">
            <a href="/" id="logo">
                <i class="logo"></i>
                <span class="site-title">sqw workplace</span>
            </a>
            <nav id="main-nav">
                
                    <a class="main-nav-link" href="/">首页</a>
                
                    <a class="main-nav-link" href="/archives">归档</a>
                
                    <a class="main-nav-link" href="/categories">分类</a>
                
                    <a class="main-nav-link" href="/tags">标签</a>
                
                    <a class="main-nav-link" href="/about">关于</a>
                
            </nav>
            
            <div id="search-form-wrap">

    <form class="search-form">
        <input type="text" class="ins-search-input search-form-input" placeholder="Search">
        <button type="submit" class="search-form-submit"></button>
    </form>
    <div class="ins-search">
    <div class="ins-search-mask"></div>
    <div class="ins-search-container">
        <div class="ins-input-wrapper">
            <input type="text" class="ins-search-input" placeholder="Type something...">
            <span class="ins-close ins-selectable"><i class="fa fa-times-circle"></i></span>
        </div>
        <div class="ins-section-wrapper">
            <div class="ins-section-container"></div>
        </div>
    </div>
</div>
<script>
(function (window) {
    var INSIGHT_CONFIG = {
        TRANSLATION: {
            POSTS: 'Posts',
            PAGES: 'Pages',
            CATEGORIES: 'Categories',
            TAGS: 'Tags',
            UNTITLED: '(Untitled)',
        },
        ROOT_URL: '/',
        CONTENT_URL: '/content.json',
    };
    window.INSIGHT_CONFIG = INSIGHT_CONFIG;
})(window);
</script>

<script src="/js/insight.js"></script>


</div>
        </div>
    </div>
    <div id="main-nav-mobile" class="header-sub header-inner">
        <table class="menu outer">
            <tr>
                
                    <td><a class="main-nav-link" href="/">首页</a></td>
                
                    <td><a class="main-nav-link" href="/archives">归档</a></td>
                
                    <td><a class="main-nav-link" href="/categories">分类</a></td>
                
                    <td><a class="main-nav-link" href="/tags">标签</a></td>
                
                    <td><a class="main-nav-link" href="/about">关于</a></td>
                
                <td>
                    
    <div class="search-form">
        <input type="text" class="ins-search-input search-form-input" placeholder="Search">
    </div>

                </td>
            </tr>
        </table>
    </div>
</header>

        <div class="outer">
            
            
                <aside id="sidebar">
   
        
    <div class="widget-wrap" id="categories">
        <h3 class="widget-title">
            <span>categories</span>
            &nbsp;
            <a id="allExpand" href="#">
                <i class="fa fa-angle-double-down fa-2x"></i>
            </a>
        </h3>
        
        
        
         <ul class="unstyled" id="tree"> 
                    <li class="directory">
                        <a href="#" data-role="directory">
                            <i class="fa fa-folder"></i>
                            &nbsp;
                            biosoft
                        </a>
                         <ul class="unstyled" id="tree"> 
                    <li class="directory">
                        <a href="#" data-role="directory">
                            <i class="fa fa-folder"></i>
                            &nbsp;
                            error
                        </a>
                         <ul class="unstyled" id="tree">  <li class="file"><a href="/wiki/biosoft/error/errorRecord/"></a></li>  </ul> 
                    </li> 
                     <li class="file"><a href="/wiki/biosoft/seqkit/">test</a></li>  <li class="file"><a href="/wiki/biosoft/hisat2-rsem-%E8%BF%9E%E7%94%A8/">hisat2 rsem 连用</a></li>  <li class="file"><a href="/wiki/biosoft/stringtie%E4%BD%BF%E7%94%A8/">stringtie使用</a></li>  <li class="file"><a href="/wiki/biosoft/bowtieAndBowtie2/">bowtieAndBowtie2</a></li>  <li class="file"><a href="/wiki/biosoft/ragtag%E4%BD%BF%E7%94%A8/">ragtag使用</a></li>  <li class="file"><a href="/wiki/biosoft/fastq-dump/">fastq-dump</a></li>  <li class="file"><a href="/wiki/biosoft/blast/">blast</a></li>  <li class="file"><a href="/wiki/biosoft/bwa/">bwa</a></li>  <li class="file"><a href="/wiki/biosoft/repeatMasker/">repeatMasker</a></li>  </ul> 
                    </li> 
                    
                    <li class="directory">
                        <a href="#" data-role="directory">
                            <i class="fa fa-folder"></i>
                            &nbsp;
                            codeSkill
                        </a>
                         <ul class="unstyled" id="tree">  <li class="file"><a href="/wiki/codeSkill/R/">R</a></li>  <li class="file"><a href="/wiki/codeSkill/shell%E6%8A%80%E5%B7%A7/">data.table用法</a></li>  <li class="file"><a href="/wiki/codeSkill/genoPlotR/">genoPlotR</a></li>  <li class="file"><a href="/wiki/codeSkill/pbsSystem/">pbsSystem</a></li>  <li class="file"><a href="/wiki/codeSkill/sbatchSystem/">sbatchSystem</a></li>  </ul> 
                    </li> 
                    
                    <li class="directory">
                        <a href="#" data-role="directory">
                            <i class="fa fa-folder"></i>
                            &nbsp;
                            game
                        </a>
                         <ul class="unstyled" id="tree">  <li class="file"><a href="/wiki/game/StardewValley/">StardewValley</a></li>  </ul> 
                    </li> 
                    
                    <li class="directory">
                        <a href="#" data-role="directory">
                            <i class="fa fa-folder"></i>
                            &nbsp;
                            knowledge
                        </a>
                         <ul class="unstyled" id="tree">  <li class="file"><a href="/wiki/knowledge/%E8%A1%A8%E7%9A%AE%E5%8F%91%E8%82%B2/">表皮发育</a></li>  <li class="file"><a href="/wiki/knowledge/fish/">fish</a></li>  <li class="file"><a href="/wiki/knowledge/lipidDroplet/">lipidDroplet</a></li>  <li class="file"><a href="/wiki/knowledge/pregnancy/">pregnancy</a></li>  <li class="file"><a href="/wiki/knowledge/paperArrange/">paperArrange</a></li>  <li class="file"><a href="/wiki/knowledge/statisticTest/">statisticTest</a></li>  </ul> 
                    </li> 
                    
                    <li class="directory open">
                        <a href="#" data-role="directory">
                            <i class="fa fa-folder-open"></i>
                            &nbsp;
                            pipeline
                        </a>
                         <ul class="unstyled" id="tree">  <li class="file"><a href="/wiki/pipeline/DNA%E6%AF%94%E5%AF%B9%E6%B5%81%E7%A8%8B/">DNA比对流程</a></li>  <li class="file active"><a href="/wiki/pipeline/gatk_snp%E6%B5%81%E7%A8%8B/">gatk_snp流程.md</a></li>  <li class="file"><a href="/wiki/pipeline/coverage-pipeline/">coverage_pipeline</a></li>  <li class="file"><a href="/wiki/pipeline/RNA%E6%B5%81%E7%A8%8B/"></a></li>  </ul> 
                    </li> 
                    
                    <li class="directory">
                        <a href="#" data-role="directory">
                            <i class="fa fa-folder"></i>
                            &nbsp;
                            tmp
                        </a>
                         <ul class="unstyled" id="tree">  <li class="file"><a href="/wiki/tmp/hexo-push-error/">hexo/hexo_push_error</a></li>  <li class="file"><a href="/wiki/tmp/hexo-vsCode%E4%BD%BF%E7%94%A8%E6%8A%80%E5%B7%A7/">hexo/vsCode使用技巧</a></li>  <li class="file"><a href="/wiki/tmp/hello-world/">Hello World</a></li>  </ul> 
                    </li> 
                     <li class="file"><a href="/wiki/SqwPersonal/">SqwPersonal</a></li>  </ul> 
    </div>
    <script>
        $(document).ready(function() {
            var iconFolderOpenClass  = 'fa-folder-open';
            var iconFolderCloseClass = 'fa-folder';
            var iconAllExpandClass = 'fa-angle-double-down';
            var iconAllPackClass = 'fa-angle-double-up';
            // Handle directory-tree expansion:
            // 左键单独展开目录
            $(document).on('click', '#categories a[data-role="directory"]', function (event) {
                event.preventDefault();

                var icon = $(this).children('.fa');
                var expanded = icon.hasClass(iconFolderOpenClass);
                var subtree = $(this).siblings('ul');
                icon.removeClass(iconFolderOpenClass).removeClass(iconFolderCloseClass);
                if (expanded) {
                    if (typeof subtree != 'undefined') {
                        subtree.slideUp({ duration: 100 });
                    }
                    icon.addClass(iconFolderCloseClass);
                } else {
                    if (typeof subtree != 'undefined') {
                        subtree.slideDown({ duration: 100 });
                    }
                    icon.addClass(iconFolderOpenClass);
                }
            });
            // 右键展开下属所有目录
            $('#categories a[data-role="directory"]').bind("contextmenu", function(event){
                event.preventDefault();
                
                var icon = $(this).children('.fa');
                var expanded = icon.hasClass(iconFolderOpenClass);
                var listNode = $(this).siblings('ul');
                var subtrees = $.merge(listNode.find('li ul'), listNode);
                var icons = $.merge(listNode.find('.fa'), icon);
                icons.removeClass(iconFolderOpenClass).removeClass(iconFolderCloseClass);
                if(expanded) {
                    subtrees.slideUp({ duration: 100 });
                    icons.addClass(iconFolderCloseClass);
                } else {
                    subtrees.slideDown({ duration: 100 });
                    icons.addClass(iconFolderOpenClass);
                }
            })
            // 展开关闭所有目录按钮
            $(document).on('click', '#allExpand', function (event) {
                event.preventDefault();
                
                var icon = $(this).children('.fa');
                var expanded = icon.hasClass(iconAllExpandClass);
                icon.removeClass(iconAllExpandClass).removeClass(iconAllPackClass);
                if(expanded) {
                    $('#sidebar .fa.fa-folder').removeClass('fa-folder').addClass('fa-folder-open')
                    $('#categories li ul').slideDown({ duration: 100 });
                    icon.addClass(iconAllPackClass);
                } else {
                    $('#sidebar .fa.fa-folder-open').removeClass('fa-folder-open').addClass('fa-folder')
                    $('#categories li ul').slideUp({ duration: 100 });
                    icon.addClass(iconAllExpandClass);
                }
            });  
        });
    </script>

    
    <div id="toTop" class="fa fa-angle-up"></div>
</aside>
            
            <section id="main"><article id="post-pipeline/gatk_snp流程" class="article article-type-post" itemscope itemprop="blogPost">
    <div class="article-inner">
        
        
            <header class="article-header">
                
                    <div class="article-meta">
                        
    <div class="article-category">
    	<i class="fa fa-folder"></i>
        <a class="article-category-link" href="/categories/pipeline/">pipeline</a>
    </div>

                        
                        
    <div class="article-date">
        <i class="fa fa-calendar"></i>
        <a href="/wiki/pipeline/gatk_snp%E6%B5%81%E7%A8%8B/">
            <time datetime="2022-07-16T09:01:36.000Z" itemprop="datePublished">2022-07-16</time>
        </a>
    </div>


                        
                            <i class="fa fa-bar-chart"></i>
                            <span id="busuanzi_container_site_pv"><span id="busuanzi_value_page_pv"></span></span>    
                        
                        
                            <div class="article-meta-button">
                                <a target="_blank" rel="external nofollow noopener noreferrer" href="https://github.com/zthxxx/Wiki-site/raw/writing/source/_posts/pipeline/gatk_snp流程.md"> Source </a>
                            </div>
                            <div class="article-meta-button">
                                <a target="_blank" rel="external nofollow noopener noreferrer" href="https://github.com/zthxxx/Wiki-site/edit/writing/source/_posts/pipeline/gatk_snp流程.md"> Edit </a>
                            </div>
                            <div class="article-meta-button">
                                <a target="_blank" rel="external nofollow noopener noreferrer" href="https://github.com/zthxxx/Wiki-site/commits/writing/source/_posts/pipeline/gatk_snp流程.md"> History </a>
                            </div>
                        
                    </div>
                
                
    
        <h1 class="article-title" itemprop="name">
            gatk_snp流程.md
        </h1>
    

            </header>
        
        
        <div class="article-entry" itemprop="articleBody">
        
        
            
                <div id="toc" class="toc-article">
                <strong class="toc-title">Catalogue</strong>
                    <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#imputation-and-phasing"><span class="toc-number">1.</span> <span class="toc-text">imputation and phasing</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#convert-format"><span class="toc-number">2.</span> <span class="toc-text">convert format</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#generate-tped-x2F-tfam"><span class="toc-number">3.</span> <span class="toc-text">generate tped&#x2F;tfam</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#generate-phenotype-file"><span class="toc-number">4.</span> <span class="toc-text">generate phenotype file</span></a></li></ol>
                </div>
            
        
        
            <p>gatk官方博文，描述 gatk4和之前版本的差异 <a target="_blank" rel="external nofollow noopener noreferrer" href="http://broad.io/GATK4-1">http://broad.io/GATK4-1</a><br>一些terra上的实例：<a target="_blank" rel="external nofollow noopener noreferrer" href="https://terra.bio/#library/showcase">https://terra.bio/#library/showcase</a></p>
<p>数据预处理：<br>map<br>    bam文件中会包含read group的信息，在之后的分析中要用到<br>mark duplicates<br>recalibrate base quality score（BQSR）：对machine error（包含pcr error和测序error）重新校准<br>analysis-ready bam</p>
<p>多样本可以做joint calling<br>    对每个样本单独call snp，得到 gvcf（包含基因组上所有位点的信息）<br>    用所有样本的gvcf 做一个 joint calling，筛选出至少一个样本中存在变异的位点</p>
<p>建立索引<br>nohup ~&#x2F;biosoft&#x2F;gatk-4.1.9.0&#x2F;gatk CreateSequenceDictionary -R subelongatus.fa -O gatk.dict &amp;&gt; gatkIndex.log &amp;</p>
<p>gatk必须要sam&#x2F;bam文件的头文件信息（reads group），其中ID，PL和SM信息是必须要的<br>若是忘记添加read group信息还以通过 AddOrReplaceReadGroups 添加<br>GATK 要求read group的格式<br>ID &#x3D; Read group identifier<br>　　每一个read group 独有的ID，每一对reads 均有一个独特的ID，可以自定义命名；<br>PL &#x3D; Platform<br>　　测序平台；ILLUMINA, SOLID, LS454, HELICOS and PACBIO，不区分大小写；<br>SM &#x3D; sample<br>　　reads属于的样品名；SM要设定正确，因为GATK产生的VCF文件也使用这个名字;<br>LB &#x3D; DNA preparation library identifier<br>　　对一个read group的reads进行重复序列标记时，需要使用LB来区分reads来自那条lane;有时候，同一个库可能在不同的lane上完成测序;为了加以区分，<br>　　同一个或不同库只要是在不同的lane产生的reads都要单独给一个ID. 一般无特殊说明，成对儿read属于同一库，可自定义，比如：library1</p>
<p>gatk AddOrReplaceReadGroups -I .bam -O .add.bam -LB library1 -PL illumina -PU pl1 -SM name</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br></pre></td><td class="code"><pre><span class="line">#build db</span><br><span class="line">bowtie2-build --threads 40 ./turtle.zjuv1.fa tt</span><br><span class="line">#get map.sh</span><br><span class="line">ls ../00.dnaData/*.fastq| xargs -n1 | awk &#x27;NR%2!=0&#123;print&quot;-1 &quot;$0&#125;NR%2==0&#123;print&quot;-2 &quot;$0&#125;&#x27; |xargs -n4 | while read line ;do echo &quot;bowtie2 -p 20 -x tt &quot;$line&quot; | samtools view -bS  - &gt; &quot; &gt;&gt; map.log ;done</span><br><span class="line">ls ../00.dnaData/*.fastq|xargs -n2 | cut -d&quot; &quot; -f1 | sed &#x27;s/\..*SRR/SRR/g&#x27; | sed &#x27;s/\_.*fastq//g&#x27; | awk &#x27;&#123;print$0&quot;.bam &amp;&quot;&#125;&#x27;&gt; bamname.log</span><br><span class="line">paste map.log bamname.log | awk &#x27;NR%10==0&#123;print$0&quot;\nwait&quot;&#125;NR%10!=0&#123;print$0&#125;&#x27;|sed &#x27;s/\t/ /g&#x27; &gt; map.sh</span><br><span class="line">nohup ./map.sh</span><br><span class="line">#bam input prepare</span><br><span class="line">ls *.bam | xargs -n1 | while read line ;do echo &quot;samtools view -bS &quot;$line&quot; -t ./turtle.zjuv1.fa.fai -o&quot; $line.fai.bam&quot; &amp;&quot; &gt;&gt; step1.sh;done </span><br><span class="line">./step1.sh</span><br><span class="line">ls *.fai.bam | xargs -n1 | while read line ;do echo &quot;samtools sort -@ 10 -m 10G &quot;$line&quot; -o &quot;$line.sorted&quot; &amp;&quot; &gt;&gt; step2.sh ;done</span><br><span class="line">./step2.sh</span><br><span class="line">#add head</span><br><span class="line">ls *.sorted| xargs -n1| while read line ;do echo &quot;/public1/home/yifeng/yifeng/tools/java/jdk1.8.0_271/bin/java -Xmx10G -jar /public1/home/yifeng/yifeng/tools/anaconda3/envs/py3/share/picard-2.23.8-0/picard.jar AddOrReplaceReadGroups I=/public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/&quot;$line&quot; O=/public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/&quot;$line.addhead&quot; RGLB=lib&quot;$line&quot; RGPL=ILLUMINA RGPU=unit1 RGSM=20 &amp;&quot; &gt;&gt; addhead.sh;done</span><br><span class="line">#这里RGSM=后面的20记得改 改成sample name</span><br><span class="line">./addhead.sh</span><br><span class="line">#resort</span><br><span class="line">ls *.addhead |xargs -n 1 | while read line ; do echo &quot;/public1/home/yifeng/yifeng/tools/java/jdk1.8.0_271/bin/java -Xmx50G -jar /public1/home/yifeng/yifeng/tools/anaconda3/envs/py3/share/picard-2.23.8-0/picard.jar SortSam I=/public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/&quot;$line&quot; O=/public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/&quot;$line.finalsort.bam&quot; SORT_ORDER=coordinate &amp;&quot; &gt;&gt; resort.sh;done</span><br><span class="line">#ulimit -n 65535 &amp;&amp; remember to set wait for each 5line in case memory out of range</span><br><span class="line">./resort.sh</span><br><span class="line">#mark dup</span><br><span class="line">ls *.finalsort.bam | xargs -n 1 | while read line ; do echo &quot;/public1/home/yifeng/yifeng/tools/java/jdk1.8.0_271/bin/java -Xmx50G -jar /public1/home/yifeng/yifeng/tools/anaconda3/envs/py3/share/picard-2.23.8-0/picard.jar MarkDuplicates I=/public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/&quot;$line&quot; O=/public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/&quot;$line.dup.bam&quot; METRICS_FILE=/public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/&quot;$line.metrics&quot; MAX_FILE_HANDLES=65535&quot; ;done  &gt;&gt; markdup.sh</span><br><span class="line">#rename</span><br><span class="line">ls *.dup.bam |xargs -n1 | while read line ;do echo &quot;mv &quot;$line &gt;&gt; log1 ;done</span><br><span class="line">ls *.dup.bam |xargs -n1 | sed &#x27;s/.bam.fai.*.dup/.dup/g&#x27; &gt; log2</span><br><span class="line">paste log1 log2 | awk &#x27;&#123;print$0&quot; &amp;&quot;&#125;&#x27; &gt;log3</span><br><span class="line">./log3 &amp;&amp; rm log*</span><br><span class="line">#samtools index</span><br><span class="line">ls *.dup.bam |xargs -n 1| while read line ;do echo &quot;samtools index &quot;$line&quot; &amp;&quot; &gt;&gt; bamindex.sh ;done</span><br><span class="line">./bamindex.sh</span><br><span class="line">#get dict file of ref genome</span><br><span class="line">/public1/home/yifeng/yifeng/tools/java/jdk1.8.0_271/bin/java -Xmx2G -jar /public1/home/yifeng/yifeng/tools/anaconda3/envs/py3/share/picard-2.23.8-0/picard.jar CreateSequenceDictionary R=turtle.zjuv1.fa O=/public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/turtle.zjuv1.dict</span><br><span class="line">#generate gvcf file round1</span><br><span class="line">ls *.dup.bam | xargs -n 1 | while read line ; do echo &quot;/public1/home/yifeng/yifeng/tools/java/jdk1.8.0_271/bin/java -jar /public1/home/yifeng/yifeng/tools/anaconda3/envs/py3/opt/gatk-3.8/GenomeAnalysisTK.jar -T HaplotypeCaller -R /public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/turtle.zjuv1.fa -I /public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/&quot;$line&quot; -ERC GVCF -stand_call_conf 30 -ploidy 2 -o /public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/&quot;$line.vcf.rd1&quot; -variant_index_type LINEAR -variant_index_parameter 128000 &amp;&quot; &gt;&gt; vcfround1.sh ;done</span><br><span class="line">./vcfround1.sh</span><br><span class="line">#rename samples in gvcf</span><br><span class="line">ls *.rd1 | xargs -n1 | while read line ;do echo &quot;/public1/home/yifeng/yifeng/tools/java/jdk1.8.0_271/bin/java -jar /public1/home/yifeng/yifeng/tools/anaconda3/envs/py3/share/picard-2.23.8-0/picard.jar RenameSampleInVcf I=&quot;$line&quot; O=&quot;$(basename $line .vcf.rd1).rename.vcf&quot; NEW_SAMPLE_NAME=&quot;$(basename $line .vcf.rd1)&quot; &amp;&quot; ;done  | awk &#x27;NR%5==0&#123;print$0&quot;\nwait&quot;&#125;NR%5!=0&#123;print$0&#125;&#x27;&gt; rename.sh </span><br><span class="line">./rename.sh</span><br><span class="line">#genotype</span><br><span class="line">ls *.rename.vcf  |xargs -n1 | while read line ;do echo &quot;  --variant &quot;$line ;done |xargs | while read line ;do echo &quot;/public1/home/yifeng/yifeng/tools/java/jdk1.8.0_271/bin/java -jar /public1/home/yifeng/yifeng/tools/anaconda3/envs/py3/opt/gatk-3.8/GenomeAnalysisTK.jar -T GenotypeGVCFs -R /public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/turtle.zjuv1.fa &quot;$line &quot;-o combine.output.vcf&quot; &gt;&gt; genotype.sh;done </span><br><span class="line">./genotype.sh</span><br><span class="line">#filter snps</span><br><span class="line">/public1/home/yifeng/yifeng/tools/java/jdk1.8.0_271/bin/java -jar /public1/home/yifeng/yifeng/tools/anaconda3/envs/py3/opt/gatk-3.8/GenomeAnalysisTK.jar -T VariantFiltration -R /public1/home/Zexian/turtle_proj/19.recombination_rate/01.bowtie/turtle.zjuv1.fa --variant combine.output.vcf --clusterSize 4 --clusterWindowSize 10 --filterName &quot;lowMQRankSum&quot; --filterExpression &quot;MQRankSum &lt; -12.5&quot; --filterName &quot;highFS&quot; --filterExpression &quot;FS &gt; 60.0&quot; --filterName &quot;lowReadPosRankSum&quot; --filterExpression &quot;ReadPosRankSum &lt; -8.0&quot; --filterName &quot;lowMQ&quot; --filterExpression &quot;MQ &lt; 30.0&quot; --filterName &quot;lowQD&quot; --filterExpression &quot;QD &lt; 2.0&quot; --out combine.filter.snp.vcf --genotypeFilterName &quot;lowDP&quot; --genotypeFilterExpression &quot;DP &lt; 8.0&quot; &gt; filter_snp.err</span><br><span class="line">awk &#x27;$7==&quot;PASS&quot;||$1~&quot;#&quot;&#123;print&#125;&#x27; combine.filter.snp.vcf   &gt; combine.filter.snp.final.vcf</span><br></pre></td></tr></table></figure>

<p>byLongxin</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">ref=Mhap1</span><br><span class="line">refFa=ref/$&#123;ref&#125;.fasta</span><br><span class="line">sample=male</span><br><span class="line"></span><br><span class="line">samtools faidx $refFa</span><br><span class="line">gatk CreateSequenceDictionary -R=$refFa -O=ChrAsm.dict</span><br><span class="line">gatk MarkDuplicates -I=$sample.$ref.sorted.bam -O=$sample.$ref.dedup.bam -M=$sample.$ref.m</span><br><span class="line">samtools index $sample.$ref.dedup.bam</span><br><span class="line">gatk AddOrReplaceReadGroups -I=$sample.$ref.dedup.bam -O=$sample.$ref.dedup.bam.rg.bam RGID=4$sample RGLB=lib$sample RGPU=unit1 RGSM=$sample RGPL=illumina</span><br><span class="line"></span><br><span class="line">samtools index $sample.$ref.dedup.bam.rg.bam</span><br><span class="line"></span><br><span class="line">#java -Xmx83g  -jar GenomeAnalysisTK.jar -T HaplotypeCaller -R $refFa -I $sample.$ref.dedup.bam.rg.bam --emitRefConfidence GVCF -o $sample.$ref.dedup.bam.rg.bam.g.vcf.gz -nct 12</span><br><span class="line"></span><br><span class="line">gatk --java-options &quot;-Xmx83G&quot; HaplotypeCaller -R $refFa -I $sample.$ref.dedup.bam.rg.bam --emitRefConfidence GVCF -o $sample.$ref.dedup.bam.rg.bam.g.vcf.gz -nct 12</span><br></pre></td></tr></table></figure>
<p>genotyping</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">ref=/public/home/fjnu_edu_hz/qianwei/wa/rawRef/wa_Male.hap1.fasta</span><br><span class="line">#01.CombineGVCFs</span><br><span class="line">gatk CombineGVCFs \</span><br><span class="line">-R $ref \</span><br><span class="line">--variant ./$&#123;sample1&#125;/$&#123;sample1&#125;.raw.snps.indels.g.vcf \</span><br><span class="line">--variant ./$&#123;sample2&#125;/$&#123;sample2&#125;.raw.snps.indels.g.vcf \</span><br><span class="line">--variant ./$&#123;sample3&#125;/$&#123;sample3&#125;.raw.snps.indels.g.vcf \</span><br><span class="line">-O combined.raw.g.vcf.gz</span><br><span class="line">#02.genotype</span><br><span class="line">gatk --java-options &quot;-Xmx150G&quot; GenotypeGVCFs -R $ref -V combined.raw.g.vcf.gz -O All.raw.vcf.gz</span><br><span class="line">#03.filter</span><br><span class="line">gatk --java-options &quot;-Xmx150G&quot; SelectVariants -R $ref -V All.raw.vcf.gz -selectType SNP -O ref.snp.vcf</span><br><span class="line">gatk --java-options &quot;-Xmx150G&quot; VariantFiltration -R $ref -V ref.snp.vcf --filte-expression &quot; QD &lt; 2.0 || FS &gt; 60.0 || MQRankSum &lt; -12.5 || RedPosRankSum &lt; -8.0 || SOR &gt; 3.0 || MQ &lt; 40.0 &quot; --filter-name &quot;my_snp_filter&quot; -O ref.snp.filter.vcf</span><br><span class="line">gatk --java-options &quot;-Xmx150G&quot; SelectVariants -R $ref -V ref.snp.filter.vcf --exclude-filtered -O ref.snp.final.vcf.gz</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<h1 id="imputation-and-phasing"><a href="#imputation-and-phasing" class="headerlink" title="imputation and phasing"></a>imputation and phasing</h1><p>for i in {1..13}<br>do<br>vcftools –vcf ref.snp.final.vcf –chr Chr${i}_hap1  –maf 0.05 –recode –recode-INFO-all –out chr${i}</p>
<p>java -Xmx100G -jar &#x2F;public&#x2F;home&#x2F;fjnu_edu_hz&#x2F;qianwei&#x2F;biosoft&#x2F;beagle.22Jul22.46e.jar gt&#x3D;chr${i}.recode.vcf out&#x3D;chr${i}_snp.IM nthreads&#x3D;4</p>
<p>gunzip chr${i}_snp.IM.vcf.gz</p>
<p>awk ‘$1<del>&#x2F;#&#x2F; || $5!</del>&#x2F;,&#x2F;‘ chr${i}_snp.IM.vcf &gt; chr${i}_snp.IM.bi.vcf</p>
<p>shapeit -V chr${i}_snp.IM.bi.vcf –window 0.5 -O chr${i}_snp.IM.Ph –thread 4</p>
<p>done</p>
<p>for i in {1..13}; do cat chr${i}_snp.IM.Ph.haps &gt;&gt;snp.haps; done</p>
<h1 id="convert-format"><a href="#convert-format" class="headerlink" title="convert format"></a>convert format</h1><p>perl &#x2F;public&#x2F;home&#x2F;fjnu_edu_hz&#x2F;qianwei&#x2F;biosoft&#x2F;getsnpplink.pl snp.sample snp.haps snp.map snp.ped<br>plink –file snp –out snp –make-bed –allow-extra-chr –allow-no-sex</p>
<h1 id="generate-tped-x2F-tfam"><a href="#generate-tped-x2F-tfam" class="headerlink" title="generate tped&#x2F;tfam"></a>generate tped&#x2F;tfam</h1><p>plink –bfile snp –chr-set 13 –allow-extra-chr –allow-no-sex –recode 12 transpose –output-missing-genotype 0 –out snp<br>plink –bfile snp –chr-set 13 –allow-extra-chr –pca –out snp</p>
<h1 id="generate-phenotype-file"><a href="#generate-phenotype-file" class="headerlink" title="generate phenotype file"></a>generate phenotype file</h1><p>awk ‘{print $1,$2,$6}’ snp.tfam &gt; snp.pheno</p>
<p>#run emmax and use snp.hBN as final result.<br>&#x2F;public&#x2F;home&#x2F;fjnu_edu_hz&#x2F;qianwei&#x2F;biosoft&#x2F;emmax&#x2F;emmax-kin-intel64 -v -h -s -d 10 snp<br>&#x2F;public&#x2F;home&#x2F;fjnu_edu_hz&#x2F;qianwei&#x2F;biosoft&#x2F;emmax&#x2F;emmax-kin-intel64 -v -h -d 10 snp<br>&#x2F;public&#x2F;home&#x2F;fjnu_edu_hz&#x2F;qianwei&#x2F;biosoft&#x2F;emmax&#x2F;emmax-intel64 -v -d 10 -t snp -p snp.pheno -k snp.hIBS.kinf -o snp.hIBS<br>&#x2F;public&#x2F;home&#x2F;fjnu_edu_hz&#x2F;qianwei&#x2F;biosoft&#x2F;emmax&#x2F;emmax-intel64 -v -d 10 -t snp -p snp.pheno -k snp.hBN.kinf -o snp.hBN</p>

            </div>
        
        <footer class="article-footer">
        </footer>
    </div>
</article>


    
<nav id="article-nav">
    
        <a href="/wiki/pipeline/DNA%E6%AF%94%E5%AF%B9%E6%B5%81%E7%A8%8B/" id="article-nav-newer" class="article-nav-link-wrap">
            <strong class="article-nav-caption">Newer</strong>
            <div class="article-nav-title">
                
                    DNA比对流程
                
            </div>
        </a>
    
    
        <a href="/wiki/game/StardewValley/" id="article-nav-older" class="article-nav-link-wrap">
            <strong class="article-nav-caption">Older</strong>
            <div class="article-nav-title">StardewValley</div>
        </a>
    
</nav>





    
    




<!-- baidu url auto push script -->
<script type="text/javascript">
    !function(){var e=/([http|https]:\/\/[a-zA-Z0-9\_\.]+\.baidu\.com)/gi,r=window.location.href,o=document.referrer;if(!e.test(r)){var n="//api.share.baidu.com/s.gif";o?(n+="?r="+encodeURIComponent(document.referrer),r&&(n+="&l="+r)):r&&(n+="?l="+r);var t=new Image;t.src=n}}(window);
</script>     
</section>
        </div>
        <footer id="footer">
    <div class="outer">
        <div id="footer-info" class="inner">
            sqw &copy; 2022 
            <a rel="external nofollow noopener noreferrer" target="_blank" href="http://creativecommons.org/licenses/by-nc-nd/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-nd/4.0/80x15.png"></a>
            <br> Powered by <a href="http://hexo.io/" target="_blank" rel="external nofollow noopener noreferrer">Hexo</a>. Theme - <a target="_blank" rel="external nofollow noopener noreferrer" href="https://github.com/zthxxx/hexo-theme-Wikitten">wikitten</a>
            
                <br>
                <span id="busuanzi_container_site_pv"><i class="fa fa-eye"></i> <span id="busuanzi_value_site_pv"></span></span>
                &nbsp;|&nbsp;
                <span id="busuanzi_container_site_pv"><i class="fa fa-user"></i> <span id="busuanzi_value_site_uv"></span></span>
            
        </div>
    </div>
</footer>

        

    
        
<script src="/libs/lightgallery/js/lightgallery.min.js"></script>

        
<script src="/libs/lightgallery/js/lg-thumbnail.min.js"></script>

        
<script src="/libs/lightgallery/js/lg-pager.min.js"></script>

        
<script src="/libs/lightgallery/js/lg-autoplay.min.js"></script>

        
<script src="/libs/lightgallery/js/lg-fullscreen.min.js"></script>

        
<script src="/libs/lightgallery/js/lg-zoom.min.js"></script>

        
<script src="/libs/lightgallery/js/lg-hash.min.js"></script>

        
<script src="/libs/lightgallery/js/lg-share.min.js"></script>

        
<script src="/libs/lightgallery/js/lg-video.min.js"></script>

    
    
        
<script src="/libs/justified-gallery/jquery.justifiedGallery.min.js"></script>

    
    
        <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [ ["$","$"], ["\\(","\\)"] ],
            skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code'],
            processEscapes: true,
            TeX: {
                equationNumbers: {
                  autoNumber: 'AMS'
                }
            }
        }
    });
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax();
        for (var i = 0; i < all.length; ++i)
            all[i].SourceElement().parentNode.className += ' has-jax';
    });
</script>
<script async src="//cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    



<!-- Custom Scripts -->

<script src="/js/main.js"></script>


    </div>
</body>
</html>